File: GSA/src/main/java/com/projetb32/GSA/entity/Fiche.java

package com.projetb32.GSA.entity;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.time.format.DateTimeFormatter;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Fiche implements Serializable {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id ;


    private String nresponsable;
    private String incerticide;
    private String nencadreur;
    private String observations;

    private String harrive;
    private String hdepart;


    @ManyToOne
    @JoinColumn(name = "calendar_id")
    private Calendar calendar ;

    @ManyToOne
    @JoinColumn(name = "users_id")
    private User user ;

    @ManyToOne
    @JoinColumn(name = "campany_id")
    private Company company ;


}

File: AngularJwtAuth/src/app/fiche/fiche.ts
export class fiche {


    locaux: string;
    nresponsable: string;
    incerticide: string;
    nencadreur: string;
    observations: string;

    harrive: string;
    hdepart: string;


  constructor(locaux: string, nresponsable: string, incerticide: string, nencadreur: string,
              observations: string, harrive: string, hdepart: string)
  {
    this.locaux= locaux;
    this.nresponsable= nresponsable;
    this.incerticide=incerticide;
    this.nencadreur=nencadreur;
    this.observations= observations;
    this.harrive= harrive;
    this.hdepart= hdepart;
  }
}
